src/Profiling.cpp

```diff
@@ -119,10 +120,28 @@ class InjectProfiling : public IRMutator {
     bool profiling_memory = true;
 
     // Strip down the tuple name, e.g. f.0 into f
-    string normalize_name(const string &name) {
-        vector<string> v = split_string(name, ".");
-        internal_assert(!v.empty());
-        return v[0];
+    string normalize_name(const string &name) const {
+        size_t idx = name.find('.');
+        if (idx != std::string::npos) {
+            internal_assert(idx != 0);
+            return name.substr(0, idx);
+        } else {
+            return name;
+        }
+    }
+
+    const Function *lookup_function(const string &name) const {
```

Comment:
Function is already a pointer type, so this should probably just return a Function by value. Alternatively you could return a `const Function &`, but I'm not sure how to satisfy the compiler in the case where you need to return something valid after the internal_error. Maybe just return env.begin()->second?

Reply:
I first had a reference, but I in case the lookup fails, I need to satisfy the compiler and give it a nullptr. Idk if we can easily create a nullptr-Function& instead of a Function* being nullptr?

Reply:
There is a default constructor I see. So for this unreachable code, a simple `return {};` looks a bit nicer I think?

Reply:
Isn't that returning a reference to a local? Given that it's unreachable, whatever makes the compiler shut up is fine.

Reply:
> Isn't that returning a reference to a local?
AFAICT, no. It's invoking the default constructor (the one literally defined with ` = default;`), which just initialized the `FunctionPtr` inside to be default-constructed, meaning that there are just some pointers put to `nullptr` I assume.